gammacat/webpage.py

```diff
@@ -114,6 +114,9 @@ def make_source_detail_page(self, source):
         # to the template render
         for resource in resources.resources:
             resource.url = urllib.parse.quote(resource.location)
+            resource.source_str = str('{:06d}'.format(resource.source_id))
```

Comment:
This is a little bit hacky and there may be a better way to implement this.

Reply:
I think in many cases you can do formatting in the template.
Please read the Jinja template docs, especially this:
http://jinja.pocoo.org/docs/dev/templates/#format

There will still be some cases where things are pre-computed or pre-formatted before passing it to the template, and no hard line when to do which. It's up to you when you write it, with the goal to have it as simple as possible.

Reply:
I played around with the formatting thing in python and jinja but I did not find a good solution, hence, I would like to leave it as it is.
It works and that's fine :-)